ktor-client/ktor-client-tests/common/test/io/ktor/client/tests/HeadersTest.kt

```diff
@@ -90,4 +93,15 @@ class HeadersTest : ClientLoader() {
             assertEquals(expected, message)
         }
     }
+
+    @Test
+    fun testRequestHasContentLength() = clientTests {
+        test { client ->
+            val first = client.put<String>("$TEST_SERVER/headers")
+            assertEquals("0", first)
+
+            val second = client.post<String>("$TEST_SERVER/headers")
+            assertEquals("0", second)
```

Comment:
Can't we test GET and HEAD here as well to ensure that we don't generate `Content-Length` for these methods?

Reply:
Sure, will add